src/runtime/hexagon_host.cpp

```diff
@@ -328,6 +328,20 @@ WEAK void halide_hexagon_finalize_kernels(void *user_context, void *state_ptr) {
 
 namespace {
 
+ALWAYS_INLINE ion_device_handle *uint64_to_ion_device_handle(const uint64_t &dev) {
+    static_assert(sizeof(ion_device_handle *) <= sizeof(uint64_t), "");
+    ion_device_handle *h = nullptr;
+    memcpy(&h, &dev, sizeof(dev));
```

Comment:
Isn't this stomping memory if sizeof(void*) < sizeof(dev)?

Reply:
sizeof(foo&) == sizeof(foo), so this would only stomp if sizeof(uint64_t) < sizeof(ion_device_handle*), which can't be the case for any existing hexagon device. (see also the static_assert.)

Reply:
...dammit, yes. (Another reason memcpy for this is bad.)